plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/AcidTransactionOwner.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore;
+
+import static java.util.Objects.requireNonNull;
+
+// this is just a type safe wrapper to make the APIs cleaner
```

Comment:
With this in mind, I would definitely add the ID to this class, and call it `TransactionInfo` or something like that.

Reply:
I have this another branch for later :)